pkg/operator/v1helpers/args.go

```diff
@@ -0,0 +1,103 @@
+package v1helpers
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// shellEscapePattern determines if a string should be enclosed in single quotes
+// so that it can safely be passed to shell command line.
+var shellEscapePattern *regexp.Regexp
+
+func init() {
+	// some characters have special meaning to the shell and may need to be escaped
+	// the following regexp defines safe characters that don't require escaping
+	shellEscapePattern = regexp.MustCompile(`[^\w@%+=:,./-]`)
+}
+
+// FlagsFromUnstructuredEscaped process the given arguments and determines if they should be escaped so that they can be safely passed as shell command line arguments.
+// unstructuredArgs holds unprocessed arguments usually retrieved from an operator's configuration file under a specific key.
+//
+// Use ToFlagSlice function to get a slice of string flags.
+func FlagsFromUnstructuredEscaped(unstructuredArgs map[string]interface{}) (map[string][]string, error) {
+	return flagsFromUnstructured(unstructuredArgs, true)
+}
+
+// FlagsFromUnstructured process the unstructured arguments usually retrieved from an operator's configuration file under a specific key.
+// There are only two supported/valid types for arguments, that is []sting and/or string.
+// Passing a different type yeld an error.
+//
+// Use ToFlagSlice function to get a slice of string flags.
+func FlagsFromUnstructured(unstructuredArgs map[string]interface{}) (map[string][]string, error) {
+	return flagsFromUnstructured(unstructuredArgs, false)
+}
+
+// ToFlagSlice transforms the provided arguments to a slice of string flags.
+// A flag name is taken directly from the key and the value is simply attached.
+// A flag is repeated iff it has more than one value.
+func ToFlagSlice(args map[string][]string) []string {
+	var keys []string
+	for key := range args {
```

Comment:
won't `keys := append([]string{}, args...)` do the same?

Reply:
`args` is a map, we need to iterate to get the keys.